gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/Field.java

```diff
@@ -0,0 +1,370 @@
+    }
+
+    /**
+     * Returns an {@link Value#INTEGER} field value.
+     */
+    public static Type integer() {
+      return new Type(Value.INTEGER);
+    }
+
+    /**
+     * Returns a {@link Value#FLOAT} field value.
+     */
+    public static Type floatingPoint() {
+      return new Type(Value.FLOAT);
+    }
+
+    /**
+     * Returns a {@link Value#BOOLEAN} field value.
+     */
+    public static Type bool() {
+      return new Type(Value.BOOLEAN);
+    }
+
+    /**
+     * Returns a {@link Value#TIMESTAMP} field value.
+     */
+    public static Type timestamp() {
+      return new Type(Value.TIMESTAMP);
+    }
+
+    /**
+     * Returns a {@link Value#RECORD} field value with associated list of sub-fields.
+     */
+    public static Type record(Field... fields) {
+      return new Type(Value.RECORD, ImmutableList.copyOf(fields));
+    }
+
+    /**
+     * Returns a {@link Value#RECORD} field value with associated list of sub-fields.
+     */
+    public static Type record(List<Field> fields) {
+      return new Type(Value.RECORD, ImmutableList.copyOf(checkNotNull(fields)));
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("value", value)
+          .add("fields", fields)
+          .toString();
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(value, fields);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (!(obj instanceof Type)) {
+        return false;
+      }
+      Type other = (Type) obj;
+      return Objects.equals(value, other.value)
+          && Objects.equals(fields, other.fields);
+    }
+  }
+
+  /**
+   * Mode for a BigQuery Table field. {@link Mode#NULLABLE} fields can be set to {@code null},
+   * {@link Mode#REQUIRED} fields must be provided. {@link Mode#REPEATED} fields can contain more
+   * than one value.
+   */
+  public enum Mode {
+    NULLABLE, REQUIRED, REPEATED
+  }
+
+  private final String name;
+  private final Type type;
+  private final Mode mode;
+  private final String description;
+
+  public static final class Builder {
+
+    private String name;
+    private Type type;
+    private Mode mode;
+    private String description;
+
+    private Builder() {}
+
+    /**
+     * Sets the field name. The name must contain only letters (a-z, A-Z), numbers (0-9), or
+     * underscores (_), and must start with a letter or underscore. The maximum length is 128
+     * characters.
+     */
+    public Builder name(String name) {
+      this.name = checkNotNull(name);
+      return this;
+    }
+
+    /**
+     * Sets the value of the field.
+     *
+     * @see <a href="https://cloud.google.com/bigquery/preparing-data-for-bigquery#datatypes">
+     *     Data Types</a>
+     */
+    public Builder type(Type type) {
+      this.type = checkNotNull(type);
+      return this;
+    }
+
+    /**
+     * Sets the mode of the field. By default {@link Mode#NULLABLE} is used.
```

Comment:
Fine as is (and I don't remember what we typically say, so preference is for consistency) but another way to say it would be "Sets the mode of the field. When not specified {@lilnk Mode#Nullable} will be used."